crates/sui-framework/sources/tx_context.move

```diff
@@ -88,10 +88,10 @@ module sui::tx_context {
     fun dummy_tx_hash_with_hint(hint: u64): vector<u8> {
         let tx_hash = vector[];
         let i = 1;
-        let hash_length = (TX_HASH_LENGTH as u8);
+        let hash_length = TX_HASH_LENGTH;
         while (i <= hash_length) {
-            let value = if (i <= 8) ((hint >> (64 - (8 * i))) as u8) else 0;
-            vector::push_back(&mut tx_hash, value);
+            let value = i ^ hint % 256;
```

Comment:
Not a big deal since it's test only but why this change?

Reply:
I don't know, can you explain this change?

Reply:
This was crashing with an arithmetic error--I think `hint` can be > 256 and the cast fails. I don't really understand the previous math, but I think all this code really needs to do is produce a different tx hash for each `hint` (which this fix will accomplish).